backend/app/main/views.py

```diff
@@ -59,143 +88,354 @@ def get_mentor_by_email(email):
+    )
+    # Convert to JSON
+    response_json = response.json()
+
+    # Validation check
+    if (response.status_code // 100) != 2:
+        return response_json["error"], response.status_code
+
+    # Get object's parameters
+    id_number = response_json["id"]
+    name = response_json["fields"].get("Name")
+    notes = response_json["fields"].get("Notes")
+    email = response_json["fields"].get("Donor Email")
+    donations = response_json["fields"].get("Total Donated")
+    if name is None or email is None:
+        return "There is no donor with this id. Please try again.", 422
+
+    #Create and return object    
+    d = Donor(name=name, email=email, id_number=id_number, notes=notes, total_donated=donations)
+    return jsonify((d.serialize())), 200
+
+
+# Get a donor from Airtable using donor's email 
+def get_a_donor_from_email(email): 
+    response = requests.get(
+        "https://api.airtable.com/v0/appw4RRMDig1g2PFI/Donors?filterByFormula=SEARCH('{}'".format(email) + ", {Donor Email})", 
+        headers={"Authorization": str(os.environ.get("API_KEY"))},
+    )
+    # Convert to JSON
+    response_json = response.json()
+    # Validation check
+    if (response.status_code // 100) != 2:
+        return response_json["error"], response.status_code
         
+    error = handleEmailResponse(response_json["records"])
+    if(error is not None):
+        return error, 422
 
+    # Get object's parameters
+    id_number = response_json["records"][0]["id"]
+    name = response_json["records"][0]["fields"].get("Name")
+    notes = response_json["records"][0]["fields"].get("Notes")
+    email = response_json["records"][0]["fields"].get("Donor Email")
+    donations = response_json["records"][0]["fields"].get("Total Donated")
+    if name is None or email is None:
+        return "There is no donor with this email. Please try again.", 422 
 
+    #Create and return object    
+    d = Donor(name=name, email=email, id_number=id_number, notes=notes, total_donated=donations)
+    return jsonify((d.serialize())), 200
 
 
+def handleEmailResponse(res):
+    if (len(res)) == 0:
+        return "No records in this database."
+    elif (len(res)) > 1:
+        errorMsg = "The email address is associated to " + str(len(res)) + " names. It appears for"
+        for i in range(len(res)):
+            if(i == len(res) - 1):
+                errorMsg += " and " + res[i]["fields"].get("Name") + "."
+            else:
+                errorMsg += " " + res[i]["fields"].get("Name") + ","
+        return errorMsg
 
+def repeat_pagination(response_json, userRole, getResponse):
+    while 'offset' in response_json:
+        offset = response_json["offset"]	
+        response = requests.get(	
+        ("https://api.airtable.com/v0/appw4RRMDig1g2PFI/"+userRole+"?offset={}").format(offset),	
+        headers={"Authorization": str(os.environ.get("API_KEY"))},	
+        )	
+        response_json = response.json()	
+        getResponse()
```

Comment:
Added this for pagination and included it in get routes that get all mentors, clients, volunteers or donors.
For the other routes, then response is supposed to be unique.